bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/internal/RESTActivator.java

```diff
@@ -21,29 +28,69 @@
 
     private final Logger logger = LoggerFactory.getLogger(RESTActivator.class);
 
-	private static BundleContext context;
-	
-	/**
-	 * Called whenever the OSGi framework starts our bundle
-	 */
-	public void start(BundleContext bc) throws Exception {
-		context = bc;
-		logger.debug("REST API has been started.");
-	}
-
-	/**
-	 * Called whenever the OSGi framework stops our bundle
-	 */
-	public void stop(BundleContext bc) throws Exception {
-		context = null;
-		logger.debug("REST API has been stopped.");
-	}
-	
-	/**
-	 * Returns the bundle context of this bundle
-	 * @return the bundle context
-	 */
-	public static BundleContext getContext() {
-		return context;
-	}
+    private static BundleContext context;
+
+    private static final String JAXRS_CONNECTOR_CONFIG = "com.eclipsesource.jaxrs.connector";
+
+    private static final String JAXRS_CONNECTOR_PROPERTY = "root";
+
+    /**
+     * Called whenever the OSGi framework starts our bundle
+     */
+    public void start(BundleContext bc) throws Exception {
+        context = bc;
+        setJaxrsConfiguration();
+        logger.debug("REST API has been started.");
+    }
+
+    /**
+     * Called whenever the OSGi framework stops our bundle
+     */
+    public void stop(BundleContext bc) throws Exception {
+        context = null;
+        logger.debug("REST API has been stopped.");
+    }
+
+    /**
+     * Returns the bundle context of this bundle
+     * 
+     * @return the bundle context
+     */
+    public static BundleContext getContext() {
+        return context;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void setJaxrsConfiguration() throws IOException {
+        ConfigurationAdmin configurationAdmin = getConfigurationAdmin();
+
+        if (configurationAdmin != null) {
+            Configuration configuration = configurationAdmin.getConfiguration(JAXRS_CONNECTOR_CONFIG);
+
+            if (configuration != null) {
+                Dictionary properties = configuration.getProperties();
+
+                if (properties == null) {
+                    properties = new Properties();
+                }
+
+                String rootAlias = (String) properties.get(JAXRS_CONNECTOR_PROPERTY);
+                if (!MDNSAnnouncer.REST_SERVLET_ALIAS.equals(rootAlias)) {
```

Comment:
Maybe I can create a utility constants class where all the configuration constants can be stored?